src/pages/storage/StorageVolumeDetail.tsx

```diff
@@ -62,33 +64,37 @@ const StorageVolumeDetail: FC = () => {
   }
 
   return (
-    <main className="l-main">
-      <div className="p-panel">
-        <StorageVolumeHeader volume={volume} project={project} />
-        <div className="p-panel__content storage-volume-form">
-          <NotificationRow />
-          <Row>
-            <TabLinks
-              tabs={tabs}
-              activeTab={activeTab}
-              tabUrl={`/ui/project/${project}/storage/detail/${pool}/${type}/${volume.name}`}
-            />
+    <CustomLayout
+      header={<StorageVolumeHeader volume={volume} project={project} />}
+      contentClassName="sotrage-volume-form"
+    >
+      {activeTab !== "snapshots" && <NotificationRow />}
```

Comment:
The current behaviour with instance detail page is that snapshot related notifications appear below the tab links but everything else appear above. This is done in `InstanceSnapshots.tsx` with the `inTabNotification` state and the `NotificationRowLegacy` component. If we change this behaviour here then we would have to change it in all the other detail pages e.g. instance detail, network detail etc. Should we do that?

Reply:
The idea for notification location was: Buttons above the tab list should produce notifications above them as well. In the instance detail page we have the state actions (start, stop, etc).

For buttons below the tab list, the notification should appear below that list. I think there are no actions for volumes above the tab list that trigger a notification. So it seemed, we can move the NotificationRow down and rely on it from all tab-conten-components. I think we don't have to change the other detail pages for this.

Reply:
Actually, I am not too happy about the in-tab notification implementation in instance detail. It couples the components very much for message passing. I hope it is something we'll improve very soon with the notification centre initiative.

Reply:
> The idea for notification location was: Buttons above the tab list should produce notifications above them as well. In the instance detail page we have the state actions (start, stop, etc).
> 
> For buttons below the tab list, the notification should appear below that list. I think there are no actions for volumes above the tab list that trigger a notification. So it seemed, we can move the NotificationRow down and rely on it from all tab-conten-components. I think we don't have to change the other detail pages for this.

I think one example is that if you go to the configurations tab, edit something for an instance or volume, the update notification appears above the tab list. This is because the `NotificationRow` component sits above `TabLinks` in `InstanceDetail`. This goes against the logic just described since the action originates from below the tab list. The problem now is that if we move `NotificationRow` below `TabLinks` in `InstanceDetail`, then notifications for instance actions above the tab list will also move below the tabs. Now looking at it, I think I understand why the snapshot notifications was done internal to the snapshots tab component.